blueos_repository/consolidate.py

```diff
@@ -0,0 +1,151 @@
+REPO_ROOT = "https://raw.githubusercontent.com/bluerobotics/BlueOS-Extensions-Repository/master/repos"
+
+
+class EnhancedJSONEncoder(json.JSONEncoder):
+    def default(self, o):  # type: ignore
+        if dataclasses.is_dataclass(o):
+            return dataclasses.asdict(o)
+        return super().default(o)
+
+
+@dataclasses.dataclass
+class Author:
+    name: str
+    email: str
+
+    @staticmethod
+    def from_json(json_dict: Dict[str, str]) -> "Author":
+        return Author(name=json_dict["name"], email=json_dict["email"])
+
+
+@dataclasses.dataclass
+class Company:
+    name: str
+    about: Optional[str]
+    email: Optional[str]
+
+    @staticmethod
+    def from_json(json_dict: Dict[str, str]) -> "Company":
+        return Company(name=json_dict["name"], email=json_dict.get("email", None), about=json_dict.get("about", None))
+
+
+# pylint: disable=too-many-instance-attributes
+@dataclasses.dataclass
+class Version:
+    permissions: Optional[Dict[str, Any]]
+    requirements: Optional[str]
+    tag: Optional[str]
+    website: str
+    authors: List[Author]
+    docs: Optional[str]
+    readme: Optional[str]
+    company: Optional[Company]
+    support: Optional[str]
+
+
+@dataclasses.dataclass
+class RepositoryEntry:
+    identifier: str
+    name: str
+    description: str
+    docker: str
+    versions: Dict[str, Version]
+    extension_logo: Optional[str]
+    company_logo: Optional[str]
+
+
+class Consolidator:
+    registry = Registry()
+    consolidated_data: List[RepositoryEntry] = []
+
+    @staticmethod
+    async def fetch_readme(url: str) -> str:
+        if not url.startswith("http"):
+            print(f"Invalid Readme url: {url}")
+            return ""
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url) as resp:
+                if resp.status != 200:
+                    print(f"Error status {resp.status}")
+                    raise Exception(f"Could not get readme {url}")
+                return await resp.text()
+
+    async def all_repositories(self) -> AsyncIterable[RepositoryEntry]:
+        repos = Path("./repos")
+        for repo in repos.glob("**/*.json"):
+            with open(repo, "r", encoding="utf-8") as individual_file:
+                company, extension_name = repo.as_posix().split("/")[1:3]
+                identifier = ".".join([company, extension_name])
+                try:
+                    data = json.load(individual_file)
+                except Exception as exc:
+                    raise Exception(f"Unable to parse file {repo}") from exc
+                company_logo = (repo / "../../company_logo.png").resolve().relative_to(repos.resolve())
+                extension_logo_file = (repo / "../extension_logo.png").resolve()
+                if extension_logo_file.exists():
+                    extension_logo = extension_logo_file.resolve().relative_to(repos.resolve())
+                else:
+                    extension_logo = company_logo
```

Comment:
don't we need to enforce the company logo ?

Reply:
Yep, I'm asking if we should _enforce_ the company logo.
We can also add it in a CI event for automatic review.

Reply:
Does every company/developer need to have and provide a company logo? It seems _useful_ if they do, but maybe it's ok to not provide one? If you want it to be a bit more obvious that each company is unique we could auto-generate an image based on a hash of the name or something instead of using the same default for everyone.

Reply:
I like the idea of generating something, maybe just the initials as in slack? that can be in another pr though

Reply:
I dont think we need to enforce a logo, we can generate one automatically, maybe